File: Project_Sourcecode_java/src/main/java/edu/iit/cs586/project/output/strategy/MakeWithdraw/MakeWithdraw2.java
package edu.iit.cs586.project.output.strategy.MakeWithdraw;

import edu.iit.cs586.project.datastore.DS2;
import edu.iit.cs586.project.datastore.DataStore;

/**
 * @author CharmingLee <charminglee@jieztech.com>
 * @date 2022/5/2 23:26
 */
public class MakeWithdraw2 implements MakeWithdraw{
    @Override
    public void makeWithdraw(DataStore ds) {
        float b = ((DS2)ds).b;
        float w = ((DS2)ds).temp_w;
        ((DS2)ds).b = b - w;
    }
}

File: Project_Sourcecode_go/output/strategy/makeWithdraw/MakeWithdraw2.go
package makeWithdraw

import (
	"cs586_project/datastore"
	"fmt"
)

type MakeWithdraw2 struct {
}

func (s *MakeWithdraw2) MakeWithdraw(ds interface{}) {
	ds2 := ds.(*datastore.DS2)
	b := ds2.Balance
	w := ds2.Temp_w
	ds2.Balance = b - w
	fmt.Printf("Make Withdraw: %v\n", w)
}
